src/calculated-property-rule.ts

```diff
@@ -32,14 +32,13 @@ export class CalculatedPropertyRule extends Rule {
 			if (options.property) {
 				property = typeof options.property === "string" ? rootType.getProperty(options.property) as Property : options.property as Property;
 
-				if (!options.isDefaultValue) {
-					// indicate that the rule is responsible for returning the value of the calculated property
-					options.returns = [property];
-				}
-				else {
+				if (options.isDefaultValue) {
 					// Ensure the default value rule runs on init of a new instance
 					(options as RuleInvocationOptions).onInitNew = true;
 				}
+
+				// indicate that the rule is responsible for returning the value of the calculated property
+				options.returns = [property];
```

Comment:
The way I'm reading this, default value calculations would have never run for existing entities? Or, at least not unless one of their inputs changed. So, with this change, they will also run on first access if needed.

Reply:
You're correct. Previously, default calculations would only run when a dependency changed for an existing entity.